gulpfile.js

```diff
@@ -2,26 +2,65 @@
 
 (function (pkg, gulp,
            jshint, jscs, stylish, mocha, connect, less, requirejs, uglify, replace, rename, header, awspublish,
-           aws, yargs, rimraf, mkdirp, recursive) {
+           aws, yargs, fs, path, rimraf, mkdirp, recursive) {
     'use strict';
 
-    var environment = yargs.argv.env || 'development',
-        environmentPrefix = {
-            staging: 'staging-',
-            uat: 'uat-',
-            production: ''
-        },
-        headerTemplate =
-            '/**\n' +
-            ' * ${pkg.name}\n' +
-            ' * ${pkg.description}\n' +
-            ' * \n' +
-            ' * @version ${pkg.version}\n' +
-            ' * @date ' + (new Date()).toISOString() + '\n' +
-            ' * @environment ' + environment + '\n' +
-            ' * @link ${pkg.homepage}\n' +
-            ' * @license ${pkg.license}\n' +
-            ' */\n\n';
+    var environment, environmentPrefix, headerTemplate, autodiscoverTests;
+
+    environment = yargs.argv.env || 'development';
+
+    environmentPrefix = {
+        staging: 'staging-',
+        uat: 'uat-',
+        production: ''
+    };
+
+    headerTemplate =
+        '/**\n' +
+        ' * ${pkg.name}\n' +
+        ' * ${pkg.description}\n' +
+        ' * \n' +
+        ' * @version ${pkg.version}\n' +
+        ' * @date ' + (new Date()).toISOString() + '\n' +
+        ' * @environment ' + environment + '\n' +
+        ' * @link ${pkg.homepage}\n' +
+        ' * @license ${pkg.license}\n' +
+        ' */\n\n';
+
+    autodiscoverTests = function (callback) {
+        var processNextPath,
+            rootPath = path.join(__dirname, 'test', 'spec'),
+            paths = [ '' ],
+            tests = [];
+        processNextPath = function () {
+            var dir;
+            if (paths.length === 0) {
+                return callback(null, tests);
+            }
+            dir = paths.pop();
+            fs.readdir(path.join(rootPath, dir), function (err, files) {
+                var processNextFile;
+                processNextFile = function () {
```

Comment:
It could but I don't like the way the IDE indents this style.  Anyway I've changed it to a better style that doesn't use `var` but uses function expressions instead.